src/Qir/Runtime/public/QSharpSimApi_I.hpp

```diff
@@ -72,15 +72,15 @@ namespace Quantum
             PauliId bases[],
             Qubit targets[],
             Result result,
-            const char* failureMessage) = 0;
+            const char* failureMessage) = 0;    // TODO: The `failureMessage` is not used, consider removing. The `bool` is returned.
```

Comment:
How come the failure message is not used? That would be the message that is given as argument in the code, isn't it? If so, it would be good to show this message to the user.

Reply:
The [current implementation](https://github.com/microsoft/qsharp-runtime/blob/a234eb8360c1add269fdcfdd115e0938a8292708/src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp#L407) of `AssertProbability()` does not use that parameter (both in FullState- and ToffoliSimulator). Just returns the bool telling whether the assert succeeded or failed. 
My code prints that message _based on that result_, _after the call_ to `AssertProbability()`. See the [`AssertMeasurement.cpp`](https://github.com/microsoft/qsharp-runtime/pull/675/files#diff-c4126d94d5ba7b634159ee02ed30598aeb10f6c48bc4b1a4ad720284341ffb6bR44), line 44.

Reply:
#679